### ISSUE TITLE
`Time spent per user` metric decrease after V4->V5 update.

### ISSUE THREAD

**Issue by kekchpek** [adjust/unity_sdk#308]
After we had updated Adjust Unity SDK in our project, we noticed significant decrease of `Time spent per user` in cohort analytics dashboard. We use other analytics services like Firebase and this behaviour isn't reproduced there. Are there any changes in time tracking or something in V5 SDK, that could  affect this?
<img width="1438" alt="image" src="https://github.com/user-attachments/assets/f6e6ecb7-c6fa-4097-9acb-e90eb20261a9">

**Comment by uerceg** [comment]
Hi @kekchpek,

Thanks for the report. In theory, SDK v5 hasn't been touching any logic when it comes to way how SDK is counting time users are spending in foreground / background.

Is this something which you're observing in iOS and Android version of your app or is this specific to one of those platforms?

**Comment by kekchpek** [comment]
I can observe it only in iOS

**Comment by uerceg** [comment]
And do you see any change in number of daily active users (and similar) metric for that app?

**Comment by alkanyunus** [comment]
> No, I can't see any anomalies in daily users, revenue, retention or any other.
> 
> I noticed, that there are some things in my project, that differ from [Install Guide](https://dev.adjust.com/en/sdk/unity/?version=v5). In the project I have not disabled `Link Frameworks Statically` and added `addToAllTargets="true"` to the `Dependencies.xml` pod dependency. I made it, because I had problems with unchecking `Link Frameworks Statically`. When I do this, I receive a build error in my xCode project. There is a bunch of errors:
> 
> ```
> Undefined symbol: _GADAdLoaderAdTypeNative
> Undefined symbol: _GADAdSizeBanner
> Undefined symbol: _GADAdSizeLeaderboard
> Undefined symbol: _GADAdSizeMediumRectangle
> Undefined symbol: _GADCurrentOrientationAnchoredAdaptiveBannerAdSizeWithWidth
> Undefined symbol: _GADGetStringFromVersionNumber
> Undefined symbol: _OBJC_CLASS_$_GADAdLoader
> Undefined symbol: _OBJC_CLASS_$_GADAppOpenAd
> Undefined symbol: _OBJC_CLASS_$_GADBannerView
> Undefined symbol: _OBJC_CLASS_$_GADExtras
> Undefined symbol: _OBJC_CLASS_$_GADInterstitialAd
> Undefined symbol: _OBJC_CLASS_$_GADMediaView
> Undefined symbol: _OBJC_CLASS_$_GADMobileAds
> Undefined symbol: _OBJC_CLASS_$_GADNativeAdImageAdLoaderOptions
> Undefined symbol: _OBJC_CLASS_$_GADNativeAdView
> Undefined symbol: _OBJC_CLASS_$_GADNativeAdViewAdOptions
> Undefined symbol: _OBJC_CLASS_$_GADQueryInfo
> Undefined symbol: _OBJC_CLASS_$_GADRequest
> Undefined symbol: _OBJC_CLASS_$_GADRewardedAd
> Undefined symbol: _OBJC_CLASS_$_GADRewardedInterstitialAd
> Linker command failed with exit code 1 (use -v to see invocation)
> ```
> 
> It doesn't seem to be related to Adjust, but it makes me not able to disable `Link Frameworks Statically` because it affects other SDK's I use.
> 
> Maybe this could be connected to my issue?

Exactly the same here

**Comment by alkanyunus** [comment]
@kekchpek 
When we have to uncheck "Link Frameworks Statically" in order to work with adjust v5, we came up with the Other Linker Flags with a post process script to add "-ObjC" for only ios builds.

Adjust v5
Applovin MAX 
 - Google admob networks
 
`
    var projPath = PBXProject.GetPBXProjectPath(path);
    var proj = new PBXProject();
    proj.ReadFromFile(projPath);
    
    var targetGuid = proj.GetUnityMainTargetGuid();
    var frameworkGuid = proj.GetUnityFrameworkTargetGuid();

    // build settings
    // when static library disabled, Google Admob network throws compile error
    // So adding objc flag only to solve the issue
    proj.SetBuildProperty(targetGuid, "OTHER_LDFLAGS", "-ObjC");
`

Let me know if it's worked for you too.

**Comment by uerceg** [comment]
Hey @kekchpek,

Actually, yes, there are updates on this topic. In next couple of days we are planning to release 5.0.4 update which should eliminate the need for that step to be performed (uncheck static linking on EDM4U) and hopefully make SDK to handle any kind of setup you might have set for EDM4U iOS resolver.

I'll keep the ticket posted.

**Comment by uerceg** [comment]
We have just shipped [v5.0.4](https://github.com/adjust/unity_sdk/releases/tag/v5.0.4) which should hopefully address these issues.

Looking forward to hearing back from you if it fixed the issues on your end or not.

Cheers

**Comment by uerceg** [comment]
Hi @BogdanBobkov,

Let me double-chrck if I understood you well. Is the absence of the `-ObjC` flag a thing that's causing the issues for you and if you add the flag back, everything works fine?

**Comment by BogdanBobkov** [comment]
@uerceg You are right. With the `-ObjC` flag added, everything works correctly. I can see new 'register for remote notifications iOS' errors from Firebase after launching the application with that flag, but I assume this is an internal issue on our side.

**Comment by uerceg** [comment]
I see. I mean, is there something preventing you from adding it back? Presence of that flat should not do any harm to our SDK integration. Do you see Adjust SDK not working if you add the flag?

**Comment by uerceg** [comment]
Hey guys,

This ticket ended up having a few topics being discussed. I'll close it for the time being since there hasn't been any activity on it, but please, in case you're experiencing any of the issues you reported even with the latest Unity SDK version, feel free to comment / reopen.
